docs/docs/02-diagram_format.md

```diff
@@ -0,0 +1,376 @@
+---
+title: Diagram JSON Format
+---
+
+# JSON Format of `objects` argument in `draw` function
+
+> :warning: Dark mode may make example images harder to read.
+
+The `draw` function defined in `user_functions.ts` has an `objects` argument which is used to define the array of objects that is drawn on the canvas.
+One way to populate the `objects` argument is by using a JSON file containing the object array.
+
+## Core Attributes
+
+All drawable JSON objects include the following core attributes, unless explicitly stated otherwise:
+
+| Attribute | Type                | Required | Description                                                                                                                             |
+| --------- | ------------------- | -------- | --------------------------------------------------------------------------------------------------------------------------------------- |
+| `type`    | `string`            | Yes      | Specifies the type of object to be drawn. (e.g., `.class`, `.frame`, `int`, `str`). See [Object Types](#object-types) for more details. |
+| `id`      | `number` or `null`  | Yes      | A unique identifier for the object. Must be `null` for stack frames. May be `null` anywhere to render a blank ID box.                   |
+| `value`   | `any`               | Yes      | The content of the object to be drawn. Format depends on the [object type](#object-types).                                              |
+| `x`, `y`  | `number`            | No       | The x and y coordinates for the object on the canvas. Required only if the `automation` parameter of `draw` is `false`.                 |
+| `style`   | `object` or `array` | No       | Custom visual styling. See [Style API](04-style.md) for more details.                                                                   |
+
+---
+
+## Object Types
+
+| Type Value                            | Description       | Jump to more info               |
+| ------------------------------------- | ----------------- | ------------------------------- |
+| `.frame`                              | Stack frame       | [Stack Frames](#stack-frames)   |
```

Comment:
It's good that the order here matches the order of the sections on the page. However, I'd like you to change the order as follows:

```
`int`, `float`, `str`, `bool`, `None`
`list`, `tuple`
`set`
`dict`
`.class`
`.frame`
`.blank`, `.blank-frame`
```

Note that this is also suggesting how I want you to organize (and in some cases, combine) subsections.

Finally, I think the "Description" and "Jump to more info" columns are redundant. I would keep just "Description" but make it a link